### PR TITLE
track dvcyaml file

### PR DESCRIPTION
**Pull request by dberenbaum**
Now that `dvc.yaml` will be written to the root, this PR ensures it is git-tracked.

### CODE REVIEW

File: src/dvclive/live.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def _init_dvc_file(self) -> str:
                 return self._dvcyaml
             raise InvalidDvcyamlError
         if self._dvc_repo is not None:
-            return os.path.join(self._dvc_repo.root_dir, "dvc.yaml")
+            return os.path.relpath(os.path.join(self._dvc_repo.root_dir, "dvc.yaml"))
```
Comment (daavoo) [inline src/dvclive/live.py:196]: whi is the `relpath` needed?

Comment (dberenbaum) [reply src/dvclive/live.py:196]: Without it, we end up doing `git add /var/tmp/.../dvc.yaml`, which I guess is fine, so I can update the tests to expect that.

File: src/dvclive/live.py [hunk 2]
```diff
@@ -548,6 +548,10 @@ def end(self):
             catch_and_warn(DvcException, logger)(ensure_dir_is_tracked)(
                 self.dir, self._dvc_repo
             )
+            if self._dvcyaml:
+                catch_and_warn(DvcException, logger)(self._dvc_repo.scm.add)(
```
Comment (daavoo) [inline src/dvclive/live.py:552]: This is missing some test

**Review by daavoo — APPROVED** [review]

**Comment by dberenbaum** [comment]
Not sure why the tests are failing. Some error relating to the version.

**Comment by dberenbaum** [comment]
@skshetry WDYT? Should the failure block merging?

File: src/dvclive/live.py (hunk 2 above)
Comment (daavoo) [reply src/dvclive/live.py:563]: I meant the `scm.add`, it seems that we only test the `include_untracked` change

**Comment by skshetry** [comment]
@dberenbaum, let me try to rerun, it is probably fixed in setuptools_scm. If not, we can merge it and pin it separately. :)

**Review by skshetry — APPROVED** [review]

File: src/dvclive/live.py (hunk 2 above)
Comment (dberenbaum) [reply src/dvclive/live.py:563]: Good catch. Added it to `test_untracked_dvclive_files_inside_dvc_exp_run_are_added`.

**Pull request merged**